guides/cheatsheets/git.md

```diff
@@ -0,0 +1,305 @@
+# Git Cheatsheet
+
+![Git Meme!](assets/images/gitmeme.jpeg)
+
+## **Welcome!**
+
+Version control is the essence of engineering here at the Shed. Git is a free
+version control system that's responsible for everything GitHub related that
+happens locally on your computer. This cheatsheet features the most important
+and commonly used Git commands for easy reference.
+
+Full Git Techincal Guide can be
+[here](https://github.com/TheDataShed/EngineeringHandbook/blob/main/guides/technical-guides/git.md).
+
+## **Cheatsheet includes:**
+
+    1. Installation & GUI
+    2. Setup
+    3. Setup & Init
+    4. Stage & Snapshot
+    5. Branch & Merge
+    6. Inspect & Compare
+    7. Tracking Path Changes
+    8. Ignoring Patterns
+    9. Share & Update
+    10. Rewrite History
+    11. Temporary Commits
+
+## **Installation & GUI**
+
+With platform specific installers for Git, GitHub also provides the ease of
+staying up-to-date with the latest releases of the command line tool while
+providing a graphical user interface for day-to-day interaction, review, and
+repository synchronisation.
+
+You can download Git for Windows and Mac [here](https://git-scm.com/).
+
+If you prefer the Git Desktop client, navigate
+[here](https://desktop.github.com/).
+
+Alternative GUI programs are also available:
+
+- [SourceTree](https://www.sourcetreeapp.com/)
+- [Git Kraken](https://www.gitkraken.com/)
+- [VS Code Extension - GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens)
+
+---
+
+## **SETUP**
+
+---
+
+Configuring user information used across all local repositories.
+
+_Set a name that is identifiable you when review version history_.
+
+    git config --global user.name “[firstname lastname]”
+
+_Set an email address_.
+
+    git config --global user.email “[valid-email]”
+
+_Adding the `--local` option or not passing a config level option at all, will
+set the user.name for the current local repository._.
+
+    git config --local user.name "[First] [Last]"
+    git config --local user.email “[valid-email]”
+
+## **SETUP & INIT**
+
+Below are commands for initializing and cloning repositories.
+
+_Initialise an existing directory as a Git repository_.
+
+    git init
+
+_Retrieve an entire repository from a hosted location via URL_.
+
+    git clone [url]
+
+---
+
+## **STAGE & SNAPSHOT**
+
+---
+
+Working with snapshots and the Git staging area.
+
+_Show modified files in working directory, staged for your next commit_.
+
+    git status
+
+_Add a file as it looks now to your next commit (stage)_.
+
+    git add [file]
+
+_Add everything your next commit_.
+
+    git add .
```

Comment:
Can we remove this entirely? Always be explicit when adding things to a commit.

The above (`git add [file]`) is the only variant we should ever encourage.